client/protocols/trace.py

```diff
@@ -0,0 +1,127 @@
+"""Implements the TraceProtocol, which traces code and
+provides a Python Tutor visualization.
+"""
+
+import datetime as dt
+import logging
+import json
+
+from pytutor import generate_trace
+from pytutor import server
```

Comment:
Custom package that I wrote in order to not include the Python Tutor source in the ok-client: https://github.com/okpy/pytutor 

The abstraction isn't super great since it only assumes ok-client is going to use it - so it has stuff like `print("... To Close OK ...")` but that's only an issue for other users.